src/atoms/Button/Button.web.js

```diff
@@ -1,20 +1,469 @@
+          return 'medium';
+        }
+        return 'large';
+      default:
+        if (children) {
+          return 'medium';
+        }
+        return 'large';
+    }
+  },
+  fontSize({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'xxsmall';
+      case 'small':
+        return 'xsmall';
+      case 'medium':
+        return 'medium';
+      case 'large':
+        return 'medium';
+      default:
+        return 'medium';
+    }
+  },
+  align({ align }) {
+    switch (align) {
+      case 'left':
+        return 'flex-start';
+      case 'center':
+        return 'center';
+      case 'right':
+        return 'flex-end';
+      default:
+        return 'flex-start';
+    }
+  },
+  letterSpacing({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'medium';
+      default:
+        return 'small';
+    }
+  },
+  spaceBetween({ size, iconAlign, children }) {
+    if (isEmpty(children)) {
+      return null;
+    }
+
+    switch (size) {
+      case 'xsmall':
+        if (iconAlign === 'left') {
+          return [0, 0.5, 0, 0];
+        }
+        return [0, 0, 0, 0.5];
+      case 'small':
+        if (iconAlign === 'left') {
+          return [0, 0.5, 0, 0];
+        }
+        return [0, 0, 0, 0.5];
+      case 'medium':
+        if (iconAlign === 'left') {
+          return [0, 1, 0, 0];
+        }
+        return [0, 0, 0, 1];
+      case 'large':
+        if (iconAlign === 'left') {
+          return [0, 1, 0, 0];
+        }
+        return [0, 0, 0, 1];
+      default:
+        if (iconAlign === 'left') {
+          return [0, 1, 0, 0];
+        }
+        return [0, 0, 0, 1];
+    }
+  },
+  lineHeight({ size }) {
+    switch (size) {
+      case 'xsmall':
+        return 'xsmall';
+      case 'small':
+        return 'small';
+      case 'medium':
+        return 'medium';
+      case 'large':
+        return 'medium';
+      default:
+        return 'medium';
+    }
+  },
+  pointerEvents({ disabled }) {
+    if (disabled) {
+      return 'none';
+    }
+    return '';
+  },
+};
+
+const StyledButton = styled.button`
+  background-color: ${styles.backgroundColor};
+  border-radius: ${(props) => props.theme.spacings.xxsmall};
+  border: ${styles.border};
+  width: ${(props) => (props.block ? '100%' : '')};
+  cursor: pointer;
+  pointer-events: ${styles.pointerEvents};
+  outline: none;
+  &:hover {
+    border: ${styles.hoverBorder};
+    background-color: ${styles.hoverBackgroundColor};
+  }
+  &:focus {
+    border: ${styles.focusBorder};
+    background-color: ${styles.focusBackgroundColor};
+  }
+  &:active {
+    border: ${styles.activeBorder};
+    background-color: ${styles.activeBackgroundColor};
+  }
+`;
+
+const Button = ({
+  onClick,
+  children,
+  variant,
+  size,
+  disabled,
+  variantColor,
+  icon,
+  iconAlign,
+  align,
+  block,
+  type,
+}) => {
+  const theme = useTheme();
+
+  if (typeof children !== 'string' && typeof children !== 'undefined') {
```

Comment:
This check is slightly confusing. If type is not `string`, how can the second check ever be true?

Reply:
Children should only be string or `undefined`(icon only buttons).

What is confusing here? Should the `undefined` check be made 1st and then the string check?
or is the error message not clear?

Reply:
Yes, please. Move the `undefined` check first.